File: UVa/UVa10070.cpp
#include <iostream>

using namespace std;

int main(){
  unsigned long long y;
  while( scanf("%llu", &y) == 1 ){
    printf("Y = %llu\n\n", y );
    bool bu = false, leap = false, fest = false;
    if( y % 4 == 0 || (y % 100 != 0 && y % 400 == 0 ) ){ printf("This is leap year.\n"); bu = true; leap = true; }
    if( y % 15 == 0 ){ printf("This is huluculu festival year.\n"); fest = true; }
    if( bu && y % 55 == 0 ){ printf("This is bulukulu festival year.\n"); fest = true; }
    if( !leap && !fest ) printf("This is an ordinary year.\n");
    printf("\n"); 
  }
  return 0;
}

File: UVa/UVa10070.py
import sys

lines = sys.stdin.readlines()
i = 0
for i in range(len(lines)):
    line = lines[i]
    i += 1
    year = int(line)
    bu, leap, fest = False, False, False
    if year % 400 == 0 or year % 100 != 0 and year % 4 == 0:
        bu = True
        leap = True
        print("This is leap year.")
    if year % 15 == 0:
        fest = True
        print("This is huluculu festival year.")
    if leap and year % 55 == 0:
        fest = True
        print("This is bulukulu festival year.")
    if not leap and not fest:
        print("This is an ordinary year.")
    if not i == len(lines):
        print()
